src/Hakyll/Web/Template/Internal.hs

```diff
@@ -1,13 +1,12 @@
 --------------------------------------------------------------------------------
 -- | Module containing the template data structure
-{-# LANGUAGE DeriveDataTypeable         #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 module Hakyll.Web.Template.Internal
-    ( Template (..)
```

Comment:
Hi, I just noticed these commits by chance.
Not exposing the `Template` constructor any more does break my code :-/ I have to admit that #384 was quite a hacky hack, and "Internal" *does* mean *internal*, but I'd love to keep inspecting templates via `unTemplate`.  Can you re-enable this, please?
Moving the `Template` data type to the base module of course makes it impossible to export it from the `Internal` module. Why was this necessary?